pkg/controller/controller.go

```diff
@@ -1035,6 +1038,18 @@ func (p *csiProvisioner) Delete(volume *v1.PersistentVolume) error {
 	ctx, cancel := context.WithTimeout(context.Background(), p.timeout)
 	defer cancel()
 
+	// Verify if volume is attached to a node before proceeding with deletion
+	vaList, err := p.vaLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("failed to list volumeattachments: %v", err)
+	}
+
+	for _, va := range vaList {
+		if va.Spec.Source.PersistentVolumeName != nil && *va.Spec.Source.PersistentVolumeName == volume.Name {
+			return fmt.Errorf("persistentvolume %s is still attached to node %s", volume.Name, va.Spec.NodeName)
```

Comment:
Shouldn't this code be checking that `va.Status.Attached` is true?

As it stands, it checks for "is there a VolumeAttachment object for the PV", which is a bit broader than "is the PV attached". I don't know whether that makes a difference.

Reply:
Removed that check based on @msau42 comment - https://github.com/kubernetes-csi/external-provisioner/pull/438#discussion_r424775727
IIRC the precedent was another comment in another PR that the existence of `volumeattachment` was enough to fail the operation..

Reply:
I think I understand now why: a ControllerPublishVolume call might have failed, in which case the call must be retried or ControllerUnpublishVolume needs to be called (according to the CSI spec). So it's correct to prevent DeleteVolume even when the volume is not really attached.